File: Crio/Array/8_LongestSubstringWithoutRepeatingCharacter/LongestSubstringWithoutRepeatingCharacter.java
import java.util.*;

class LongestSubstringWithoutRepeatingCharacter {
    public int lengthOfLongestSubstring(String s) {
        // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        int longestSubstring = 0;
        int n = s.length();

        for(int i = 0 ; i < n ; ++i){
            HashSet<Character> marked = new HashSet<Character>();
            int right = i;
            while(right < n && !marked.contains(s.charAt(right))) {
                marked.add(s.charAt(right));
                right++;
            }
            longestSubstring = Math.max(longestSubstring , right - i);
        }
        return longestSubstring;
        // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
    }
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String s = in.nextLine();
        int result = new LongestSubstringWithoutRepeatingCharacter().lengthOfLongestSubstring(s);
        System.out.println(result);
    }
}

File: Crio/Array/8_LongestSubstringWithoutRepeatingCharacter/LongestSubstringWithoutRepeatingCharacter.cpp
#include "../crio/cpp/io/FastIO.hpp"
#include <bits/stdc++.h>
using namespace std;

class LongestSubstringWithoutRepeatingCharacter {
public:
    int lengthOfLongestSubstring(string s) {
        int longestSubstring = 0;

        // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        int n = s.size();

        for (int i = 0; i < n; ++i) {
            unordered_set<char> marked;
            int right = i;
            while (right < n && marked.find(s[right]) == marked.end()) {
                marked.insert(s[right]);
                right++;
            }
            longestSubstring = max(longestSubstring, right - i);
        }
        // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
        return longestSubstring;
    }
};

int main() {
    FastIO();
    string s;
    getline(cin, s);
    int result =
        LongestSubstringWithoutRepeatingCharacter().lengthOfLongestSubstring(s);
    cout << result;
    return 0;
}
